pyproject.toml

```diff
@@ -0,0 +1,76 @@
+[project]
+name = "pyxdf"
+description = "Python library for importing XDF (Extensible Data Format)"
+authors = [
+  { name = "Christian Kothe", email = "christian.kothe@intheon.io" },
+  { name = "Chadwick Boulay", email = "chadwick.boulay@gmail.com" },
+  { name = "Clemens Brunner", email = "clemens.brunner@gmail.com" }
+]
+license = {file = "LICENSE"}
+readme = "README.md"
+requires-python = ">=3.9"
+dynamic = ["version"]
+keywords = [
+  "XDF",
+  "pyxdf",
+  "LSL",
+  "Lab Streaming Layer",
+  "file format",
+  "biosignals",
+  "stream",
+]
+classifiers = [
+  # How mature is this project? Common values are
+  #   3 - Alpha
+  #   4 - Beta
+  #   5 - Production/Stable
+  "Development Status :: 4 - Beta",
+
+  # Indicate who your project is intended for
+  "Intended Audience :: Developers",
+  "Intended Audience :: Science/Research",
+  "Topic :: Scientific/Engineering",
+
+  # Pick your license as you wish (see also "license" above)
+  "License :: OSI Approved :: BSD License",
+
+  "Operating System :: Microsoft :: Windows",
+  "Operating System :: POSIX :: Linux",
+  "Operating System :: MacOS",
+
+  # Specify the Python versions you support here.
+  "Programming Language :: Python :: 3",
+  "Programming Language :: Python :: 3.9",
+  "Programming Language :: Python :: 3.10",
+  "Programming Language :: Python :: 3.11",
+  "Programming Language :: Python :: 3.12",
+]
+dependencies = [
+    "numpy>=2.0.2",
+]
+
+[project.urls]
+Repository = "https://github.com/xdf-modules/pyxdf"
+Issues = "https://github.com/xdf-modules/pyxdf/issues"
+Changelog = "https://github.com/xdf-modules/pyxdf/blob/main/CHANGELOG.md"
+
+[project.optional-dependencies]
+test = [
+    "pytest>=8.3.4",
+]
+dev = [
+    "ruff>=0.8.2",
+]
```

Comment:
I'd only create a single `dev` extra here. Or, maybe even better, you could use the `dev` dependency group (https://docs.astral.sh/uv/concepts/projects/dependencies/#dependency-groups).

Reply:
I just realized that you meant you'd like both pytest and ruff in a single group.

Sometimes the dev group can get pretty big and you don't want the test runner to import and install all those dependencies. But we aren't there yet. I can make it a single group...